rest/model/conversions.go

```diff
@@ -46,49 +46,73 @@ func conversionFn(opts convertFnOpts, generatedConversions map[typeInfo]string)
 		OutType:  opts.outType,
 		OutIsPtr: opts.outIsPtr,
 	}
-	if _, inIsPrimitive := opts.in.(*types.Basic); inIsPrimitive {
+	switch in := opts.in.(type) {
+	case *types.Basic:
 		info.InType = opts.in.String()
 		return simplePtrs(info, generatedConversions)
-	}
-	if intype, inIsPtr := opts.in.(*types.Pointer); inIsPtr {
-		value, isPrimitive := intype.Elem().(*types.Basic)
+	case *types.Pointer:
+		value, isPrimitive := in.Elem().(*types.Basic)
 		if isPrimitive {
 			info.InType = value.String()
 			info.InIsPtr = true
 			return simplePtrs(info, generatedConversions)
 		}
-		underlyingStruct, isStruct := intype.Elem().(*types.Named)
+		underlyingStruct, isStruct := in.Elem().(*types.Named)
 		if isStruct {
 			return objectTypeFnName(underlyingStruct, opts.outIsPtr)
 		}
 		return convertFuncs{}, errors.New("complex pointers not implemented yet")
-	}
-	if intype, inIsNamedType := opts.in.(*types.Named); inIsNamedType {
-		switch intype.String() {
+	case *types.Named:
+		switch in.String() {
 		case "time.Time":
 			return simplePtrs(info, generatedConversions)
 		default:
-			return objectTypeFnName(intype, opts.outIsPtr)
+			return objectTypeFnName(in, opts.outIsPtr)
 		}
-	}
-	if intype, inIsMap := opts.in.(*types.Map); inIsMap {
-		switch intype.String() {
+	case *types.Map:
+		switch in.String() {
 		case "map[string]interface{}":
 			return simplePtrs(info, generatedConversions)
 		default:
 			return convertFuncs{}, errors.New("complex maps not implemented yet")
 		}
-	}
-	if intype, inIsInterface := opts.in.(*types.Interface); inIsInterface {
-		switch intype.String() {
+	case *types.Interface:
+		switch in.String() {
 		case "interface{}":
 			return simplePtrs(info, generatedConversions)
 		default:
 			return convertFuncs{}, errors.New("non-empty interfaces not implemented yet")
 		}
+	case *types.Slice:
+		inName := cleanName(shortenPackage(opts.in.String()), containsPtr(opts.in.String()))
```

Comment:
Should this be in.String() like the other cases, or does that have a different result?

Reply:
Also I'm surprised this section doesn't generateConversionCode or anything. Does it happen somewhere else? Or is it not necessary? I think a comment here might be helpful.

Reply:
It's in a different spot. Added a comment